lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -129,13 +132,33 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
                 return JOURNEY_VCS_NOT_CORRELATED.toObjectMap();
             }
 
-            return checkForMatchingGpg45Profile(
+            List<SignedJWT> gpg45Credentials = new ArrayList<>();
+
+            for (SignedJWT credential : credentials) {
+                if (!VcHelper.isOperationalProfileVc(credential)) {
+                    gpg45Credentials.add(credential);
+                }
+            }
```

Comment:
(ditto with the helper)

Reply:
In this case I think we could filter the VC list _before_ parsing into `SignedJWT`s? - I don't think there's any need to handle non-GPG45 VCs here.

Reply:
I don't think `VcHelper.filterVCBasedOnProfileType` filters operational profiles. It currently returns either `OPERATIONAL_CRIS` or none `OPERATIONAL_CRIS`. It does not filter out the operational profiles matching `HMRC_MIGRATION_CRI`.

Reply:
Yeah, this is what I meant in the other comment... At the moment we have two different ways of identifying operational profiles:
- A VC with a `vot` matching an operational profile
- A VC from `HMRC_MIGRATION_CRI`

At the moment these are exactly equivalent, but it'd be nice to be consistent (or be clear about when we should use one vs. the other).

In this case, could you do `VcHelper.filterVCBasedOnProfileType(vcStoreItems, ProfileType.GPG45)`? I think that better reflects what you're trying to do - i.e. find all the GPG45 VCs.

Reply:
Yes, that's exactly what I did here with:

```
List<VcStoreItem> vcStoreItems =
                    VcHelper.filterVCBasedOnProfileType(
                            verifiableCredentialService.getVcStoreItems(userId), ProfileType.GPG45);
```

but it is still including the `HMRC_MIGRATION_CRI` vc in the returned list. 

Still investigating as the is not supposed to be the expected result. Maybe I am wrong.